updated to work with flask 2.2+

christopherpickering opened the pull request:
Replaced _request_ctx_stack.top with request_ctx per https://github.com/pallets/flask/pull/4682/files

christopherpickering commented:
@nickjj, hey, no, no errors, just the depreciation warning that in flask 2.3 the `_request_ctx_stack` will be removed.

nickjj commented:
Ah right.

This project currently has `Flask >= 0.8` defined. I do like the idea of removing the deprecation warning by using the "new" way for Flask 2.3 but we should figure out a solution that still works for versions < 2.3. I think you could detect the version of Flask and import either one based on the version and use `as` to ensure the import name is the same for both so the usage can remain as `req = request_ctx.request`.

What do you think?

christopherpickering commented:
Yeah, that's an idea. From the flask PR, they said this was existing all along, but I'm not sure what version of flask it was really added.

We could something like this:
`int(flask.version.split(".")[0]) < 2 and int(flask.version.split(".")[1]) < 2`

Probably there is a cleaner way?

nickjj commented:
I didn't fully test this but, but if you put this into a Python interpreter it works:

```python
import flask

from packaging import version

version.parse(flask.__version__) >= version.parse("2.2.0")
```

That condition is what you could do to require a specific package from Flask.

christopherpickering commented:
How this look @nickjj ? 

The tests run fine w/ tox, but not the stylecheck. It looks like the tox file needs updated?

jeffwidman commented:
I filed https://github.com/flask-debugtoolbar/flask-debugtoolbar/issues/184 to track bumping our version of Flask... I'm very open to doing so if /when we need it. But as @nickjj pointed out here, for this PR checking the version is a clean-backwards compatible method, so def the right way to go here.

jeffwidman commented on src/flask_debugtoolbar/__init__.py:
```diff
@@ -122,7 +131,11 @@ def _default_config(self, app):
+        if version_builder.parse(flask.__version__) >= version_builder.parse("2.2.0"):
+            req = request_ctx.request
+        else:
+            req = _request_ctx_stack.top.request
```
Oh I see, you can't just alias the package import (`from flask.globals import _request_ctx_stack as request_ctx`) because the object's API changed as well...

nickjj replied:
Yep I discovered that after the comment too.

jeffwidman commented on src/flask_debugtoolbar/__init__.py:
```diff
@@ -1,8 +1,17 @@
 from flask import Blueprint, current_app, request, g, send_from_directory, url_for
-from flask.globals import _request_ctx_stack
+
+
```
I don't work much with Python these days, but I thought style guides would suggest this as only 1 blank line rather than 2 between the other imports (or even no blank lines since it's just a conditional import and not a distinct function)? Which does `pycodestyle` prefer?

nickjj replied:
Generally `black` prefers 2 lines between functions but there's no set rule for spaces between imports and your first expressions. This is also a fun one because it's technically handling an import but it's also an expression.

Do you have a preference? I don't have `pycodestyle` installed locally.

jeffwidman replied:
I'm cool either way!

jeffwidman reviewed (approved):
LGTM, but @nickjj can you take a quick look too? If looks good to you, please merge...

nickjj commented:
I'm generally ok with it as a concept. I'm not loving repeating that version check twice but this also feels like a spot where if it were refactored and made into a general version compare function it might make things less readable in the end.

Do you want to wait for the author of the PR to potentially remove that 1 empty line before merging it @jeffwidman?

jeffwidman commented:
Thanks again!
